src/sys/fs/file_max.rs

```diff
@@ -0,0 +1,32 @@
+//! Retreive the file-max value from /proc/sys/fs/file-max
+
+use std::fs::File;
+use std::io::{Read, Error, ErrorKind, Result};
+
+/// Path to the file-max value
+static FILE_MAX_PATH: &'static str = "/proc/sys/fs/file-max";
+
+// Linux kernel uses get_max_files() which returns an unsigned long
+// see include/linux/fs.h
+
+/// Get file-max value for the current system
+pub fn file_max() -> Result<u64> {
+    // 19 is the max size of a u64
+    let mut buf = String::with_capacity(19);
```

Comment:
I've made it more like other parsers, tell me if that's what you were asking for